src/realm/sync/client.cpp

```diff
@@ -1208,10 +1212,24 @@ void SessionWrapper::on_flx_sync_progress(int64_t new_version, DownloadBatchStat
         return;
     }
     REALM_ASSERT(!m_finalized);
-    REALM_ASSERT(new_version >= m_flx_last_seen_version);
```

Comment:
shouldn't new_version always be either `m_flx_last_seen_version`, `m_flx_active_version` or `m_flx_active_version+1`?

Reply:
If a query bootstrap is in progress when a server initiated bootstrap takes place (which cancels the current query bootstrap), `m_flx_last_seen_version` will be `m_flx_active_version + 1` while `new_version` will be `m_flx_active_version` and this assertion will fail.

Reply:
it's still one of the cases I mentioned. I was proposing we update the assert, not keep the deleted one.

Reply:
the `m_flx_last_seen_version` can actually be greater than `m_flx_active_version + 1` in the case of a QUERY_ERROR response and it always starts out at 0 when the session is started until the first download message is received. Since the current validation and checking for server-initiated bootstrap logic is valid, I am going to leave this as-is.

Reply:
It seems that `m_flx_last_seen_version` should be updated when we get a query error then. `m_flx_active_version` is initialized though when the SessionWrapper is initialized. Either way, your call.